esprima.js

```diff
@@ -4023,18 +4023,20 @@
         if (match('=')) {
             lex();
             init = isolateCoverGrammar(parseAssignmentExpression);
-        } else if (id.type !== Syntax.Identifier) {
-            expect('=');
+        } else {
+            if (!(options && options.inFor) && (id.type !== Syntax.Identifier)) {
```

Comment:
On that thought, I think I'm just going to unroll the conditionals (easier to follow):
```
            if (id.type !== Syntax.Identifier) {
                if (!(options && options.inFor)) {
                    expect('=');
                }
            }
```